frontend/src/js/components/subscription/SubscriptionPage.tsx

```diff
@@ -0,0 +1,353 @@
+
+  //Fetch Billing profile
+  useEffect(() => {
+    dispatch(getCurrentCard());
+    dispatch(getUserBilling());
+  }, [dispatch]);
+
+  //Loading stripe Component
+  useEffect(() => {
+    // Make sure to call `loadStripe` outside of a component’s render to avoid recreating
+    // the `Stripe` object on every render - but don't initialize twice.
+    if (!stripePromise) {
+      import(/* webpackChunkName: "stripe" */ '@stripe/stripe-js').then(({ loadStripe }) => {
+        if (stripeAPIKey) {
+          stripePromise = loadStripe(stripeAPIKey).finally(() => setLoadingFinished(true));
+        }
+      });
+    } else {
+      const notStripePromise = new Promise(resolve => setTimeout(resolve, TIMEOUTS.debounceDefault));
+      Promise.race([stripePromise, notStripePromise]).then(result => setLoadingFinished(result !== notStripePromise));
+    }
+  }, [stripeAPIKey]);
+
+  useEffect(() => {
+    if (plan) {
+      setSelectedPlan(plan);
+    }
+
+    if (currentDeviceLimit && limit < currentDeviceLimit) {
+      setLimit(currentDeviceLimit);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [plan, currentDeviceLimit]);
+
+  useEffect(() => {
+    const newSelectedAddons: Record<AvailableAddon, boolean> = {} as Record<AvailableAddon, boolean>;
+    enabledAddons.forEach(addon => {
+      newSelectedAddons[addon.name] = addon.enabled && !isTrial;
+    });
+    setSelectedAddons(newSelectedAddons);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [enabledAddons]);
+
+  useEffect(() => {
+    if (debouncedLimit >= enterpriseDeviceCount) {
+      setContactReason(`For over ${enterpriseDeviceCount} devices, please contact for pricing.`);
+      setLimit(enterpriseDeviceCount);
+      setInputHelperText(`The maximum you can set is ${enterpriseDeviceCount} devices.`);
+    } else if (debouncedLimit < currentDeviceLimit) {
+      setLimit(currentDeviceLimit);
+      setContactReason(`If you want to reduce your device limit, please contact support@mender.io.`);
+      setInputHelperText(`Your current device limit is ${currentDeviceLimit}.`);
+    }
+    const snappedValue = Math.ceil(debouncedLimit / DIVISIBILITY_STEP) * DIVISIBILITY_STEP;
+    if (snappedValue !== limit) {
+      setLimit(snappedValue);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [debouncedLimit]);
+
+  useEffect(() => {
+    setInputHelperText(`The minimum limit for ${selectedPlan.name} is ${selectedPlan.minimalDeviceCount}`);
+  }, [selectedPlan]);
+
+  useEffect(() => {
+    if (!isOrgLoaded || selectedPlan.id === 'enterprise') {
+      return;
+    }
+    const addons = Object.entries(selectedAddons)
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .filter(([_, value]) => value)
+      .map(([key]) => ({ name: key }));
+    setIsPreviewLoading(true);
+    const order = {
+      preview_mode: 'recurring',
+      plan: selectedPlan.id,
```

Comment:
unfortunately I don't get a successful preview response when trying this locally (starting from os/basic => to professional => back to basic), unless I add the following:
```suggestion
    plan: selectedPlan.id === PLANS.os.id ? 'basic' : selectedPlan.id,
```
I don't think that's intentional on the backend side... will ping someone there 👌

Reply:
no need to ping, Alf is aware of the issue, it is solved with stripe compose config I think :)  I forgot to warn you, sorry